pkg/api/meta/restmapper.go

```diff
@@ -169,6 +169,15 @@ func (m *DefaultRESTMapper) ResourceSingularizer(resource string) (singular stri
 
 // VersionAndKindForResource implements RESTMapper
 func (m *DefaultRESTMapper) VersionAndKindForResource(resource string) (defaultVersion, kind string, err error) {
+	if parts := strings.Split(resource, "/"); len(parts) > 1 {
+		if len(parts) > 2 {
+			return "", "", fmt.Errorf("in version and kind for resource, resource %q has an invalid name", resource)
+		}
+		resource = parts[1]
+		if m.group != parts[0] {
+			return "", "", fmt.Errorf("in version and kind for resource, no resource %q is defined for group %q", resource, m.group)
```

Comment:
Same as above, though here I think "no resource %q is defined for group %q" is enough